e2e/app/base-page.ts

```diff
@@ -0,0 +1,220 @@
+import {ElementHandle, Page, Response} from 'puppeteer';
+
+export default abstract class BasePage {
```

Comment:
[action required] As discussed offline, please add some class-level comments to this base class. It really does help a lot!